### PR TITLE
Remove unreachable return of jenkins_job_trigger

### PR DESCRIPTION
**Pull request by yangyulely**
Remove unreachable return in jenkins_request_with_headers of  jenkins_job_trigger.py

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/jenkins/operators/jenkins_job_trigger.py [hunk 1]
```diff
@@ -71,7 +71,6 @@ def jenkins_request_with_headers(jenkins_server: Jenkins, req: Request) -> Jenki
         raise jenkins.TimeoutException(f"Error in request: {e}")
     except URLError as e:
         raise JenkinsException(f"Error in request: {e.reason}")
-    return None
```
Comment (pankajkoti) [inline airflow/providers/jenkins/operators/jenkins_job_trigger.py:75]: Do we need to update/correct the return type annotation?

Comment (yangyulely) [reply airflow/providers/jenkins/operators/jenkins_job_trigger.py:75]: Yeah, it never be None

**Comment by eladkal** [comment]
@yangyulely  can you rebase and resolve conflicts?

**Comment by yangyulely** [comment]
@eladkal, updated

**Pull request closed without merging**